astropy_helpers/commands/build_sphinx.py

```diff
@@ -170,18 +172,22 @@ def run(self):
         # prevents a continuous updating at the terminal, but there's no
         # apparent way around this.
         if self.warnings_returncode:
-            proc = subprocess.Popen([sys.executable],
+            proc = subprocess.Popen([sys.executable, '-c', subproccode.encode('utf-8')],
                                     stdin=subprocess.PIPE,
                                     stdout=subprocess.PIPE,
                                     stderr=subprocess.STDOUT)
-            stdo, _ = proc.communicate(subproccode.encode('utf-8'))
-            stdo = stdo.decode('utf-8')
 
-            print(stdo)
+            stdolines = []
+            with proc.stdout:
+                for line in iter(proc.stdout.readline, b''):
+                    line = line.strip(b'\n')
+                    stdolines.append(line)
+                    print(line.decode('utf-8'))
 
-            stdolines = stdo.splitlines()
+            # Poll to set proc.retcode
+            proc.poll()
```

Comment:
That's what [`Popen.wait`](https://docs.python.org/2/library/subprocess.html#subprocess.Popen.wait) is for, but otherwise good point in general.